File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoReadmeFragmentModule.java
package mb00.android.codehub.ui.repo.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.repo.interactor.RepoInteractor;
import mb00.android.codehub.ui.repo.router.RepoRouter;
import mb00.android.codehub.ui.repo.viewmodel.RepoReadmeViewModel;


@Module
public class RepoReadmeFragmentModule {

    @Provides
    RepoReadmeViewModel provideRepoReadmeViewModel(RepoInteractor interactor, RepoRouter router) {
        return new RepoReadmeViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoReadmeFragmentModule.kt
package mb00.android.codehub.ui.repo.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.repo.interactor.RepoInteractor
import mb00.android.codehub.ui.repo.router.RepoRouter
import mb00.android.codehub.ui.repo.viewmodel.RepoReadmeViewModel


@Module
class RepoReadmeFragmentModule {

    @Provides
    internal fun provideRepoReadmeViewModel(interactor: RepoInteractor, router: RepoRouter): RepoReadmeViewModel {
        return RepoReadmeViewModel(interactor, router)
    }

}
